src/main/java/duke/Command.java

```diff
@@ -0,0 +1,96 @@
+package duke;
+
+import java.util.ArrayList;
+
+public class Command {
+    private final CommandType type;
+    private String value = "";
+    private final ArrayList<Parameter> parameters;
+
+    public Command(String input) throws IllegalCommandException, IllegalParameterException {
```

Comment:
There might be too many parsing logics squeezed this Command constructor. The whole constructor method seems too long. You might want to consider refactoring the logic into few different methods, or consider other structure to parse an input into commands instead of working everything within the constructor.